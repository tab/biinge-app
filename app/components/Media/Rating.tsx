import React from "react"
import { View, Text } from "react-native"
import Icon from "react-native-vector-icons/Ionicons"

import { ratingStyles } from "styles"
import colors from "styles/colors"

type Props = {
  children: React.ReactNode
}

const RatingComponent = ({ children }: Props) => {
  return (
    <View style={ratingStyles.container}>
      <Icon name="star" size={20} color={colors.orangeYellow} />
      <Text style={ratingStyles.text}>{children}</Text>
    </View>
  )
}

export default RatingComponent
