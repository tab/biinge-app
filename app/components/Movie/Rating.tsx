import React from "react"
import { View, Image, Text } from "react-native"

import { ratingStyles, imdbLogoStyles } from "styles"

type Props = {
  children: React.ReactNode
}

const RatingComponent = ({ children }: Props) => {
  return (
    <View style={ratingStyles.container}>
      <Image
        style={imdbLogoStyles.image}
        source={require("../../assets/icons/imdb.png")}
      />
      <Text style={ratingStyles.text}>{children}</Text>
    </View>
  )
}

export default RatingComponent
