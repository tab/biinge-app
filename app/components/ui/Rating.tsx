import React from "react"
import { View, Text, StyleProp, TextStyle } from "react-native"

import Icon from "components/ui/Icon"
import { ratingStyles } from "styles"
import colors from "styles/colors"

type Props = {
  size: number
  style?: StyleProp<TextStyle>
  children: React.ReactNode
}

const RatingComponent = ({ size, style, children }: Props) => {
  return (
    <View style={ratingStyles.root}>
      <Icon name="star" color={colors.orangeYellow} size={size} />
      <Text style={[ratingStyles.text, style]}>{children}</Text>
    </View>
  )
}

RatingComponent.defaultProps = {
  size: 20,
}

export default RatingComponent
