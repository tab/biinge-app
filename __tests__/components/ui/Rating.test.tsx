import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"

import { it } from "@jest/globals"

import Rating from "components/ui/Rating"

describe("Rating", () => {
  it("renders correctly", () => {
    const tree = render(<Rating>7.3</Rating>)
    expect(tree).toMatchSnapshot()
  })
})
