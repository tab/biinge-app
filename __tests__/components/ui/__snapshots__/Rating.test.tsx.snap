// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#eec01e",
          "fontSize": 20,
        },
        undefined,
        {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    îº–
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 24,
          "fontWeight": "800",
          "lineHeight": 28,
          "marginLeft": 5,
        },
        {
          "color": "#181818",
        },
        undefined,
      ]
    }
  >
    7.3
  </Text>
</View>
`;
